include/seastar/core/sstring.hh

```diff
@@ -243,28 +243,40 @@ public:
         const char_type* it = str() + pos;
         const char_type* end = str() + size();
         const char_type* c_str = s.str();
-        const char_type* c_str_end = s.str() + s.size();
 
-	    if (c_str == c_str_end) {
-            /* see https://en.cppreference.com/w/cpp/string/basic_string/find
-             * - an empty substring is found at pos if and only if pos <= size()
-             */
-            return (pos <= size()) ? pos : npos;
+        if (pos > size()) {
+            return npos;
         }
 
-        while (it < end) {
-            auto i = it;
-            auto j = c_str;
-            while (i < end && j < c_str_end && *i == *j) {
-                i++;
-                j++;
+        const size_t len2 = s.size();
+        if (len2 == 0) {
+            return pos;
+        }
+
+        size_t len1 = end - it;
+        if (len1 < len2) {
+            return npos;
+        }
+
+        char_type f2 = *c_str;
+        do {
```

Comment:
nitpick: do/while is an esoteric replacement for the normal while() that is useful in some cases (when the condition doesn't need to be evaluated on the first run) but not in this code, where a normal while() would have sufficed.

Reply:
Yes, the LLVM's version use the normal `while()` loop, but since we have already calculated `len1` and ensured that `len1 < len2` before the loop, we can just skip the length calculation and the comparison in the first loop cycle, so I replace it with `do-while` loop.